src/components/NavigationBar/data/thunks.js

```diff
@@ -0,0 +1,31 @@
+/* eslint-disable import/prefer-default-export, no-unused-expressions */
+import { logError } from '@edx/frontend-platform/logging';
+
+import { getCourseHomeCourseMetadata } from './api';
+import {
+  fetchTabDenied,
+  fetchTabFailure,
+  fetchTabRequest,
+  fetchTabSuccess,
+} from './slice';
+
+export function fetchTab(courseId, tab, getTabData, targetUserId) {
+  return async (dispatch) => {
+    dispatch(fetchTabRequest({ courseId }));
+    try {
+      const courseHomeCourseMetadata = await getCourseHomeCourseMetadata(courseId, 'outline');
```

Comment:
For multiple uses of this component in other MEF, we need to make this hard-coded `outline` into param.